lib/core/widgets/home_bottom_navigation_bar.dart

```diff
@@ -0,0 +1,186 @@
+                      context.localization.bottomNavBarProfileText,
+                      3,
+                    ),
+                    SizedBox(width: _isCartClosed ? 50 : 110),
+                  ],
+                ),
+              ],
+            ),
+          ),
+          Positioned(
+            right: 0,
+            bottom: 45,
+            child: buildCartButton(),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget buildIcon(String iconKey, String label, int index) {
+    final isSelected = _selectedIndex == index;
+    final iconWidget = isSelected
+        ? getActiveBottomNavBarIcon(iconKey)
+        : getInactiveBottomNavBarIcon(iconKey);
+
+    final labelWidget = isSelected
+        ? GradientText(
+            text: label,
+            style: HomePageTextStyles.bottomNavBarActiveIconTextTextStyle,
+            gradient: AppGradients.purpleGradient,
+          )
+        : Text(
+            label,
+            style: HomePageTextStyles.bottomNavBarNotActiveIconTextTextStyle,
+          );
+
+    return GestureDetector(
+      onTap: () {
+        setState(() {
+          _selectedIndex = index;
+        });
+      },
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          iconWidget,
+          labelWidget,
+        ],
+      ),
+    );
+  }
+
+  Widget buildCartButton() {
+    return GestureDetector(
+      onTap: () {
+        setState(() {
+          _isCartClosed = !_isCartClosed;
+        });
+      },
+      child: Container(
+        height: 60,
+        decoration: BoxDecoration(
+          gradient: AppGradients.purpleGradient,
+          borderRadius: BorderRadius.circular(30),
+        ),
+        padding: const EdgeInsets.symmetric(
+          horizontal: 22,
+          vertical: 10,
+        ),
+        child: _isCartClosed
+            ? AppIcons.shoppingCartIcon
+            : Row(
+                children: [
+                  AppIcons.shoppingCartIcon,
+                  const SizedBox(
+                    width: 4,
+                  ),
+                  Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      Text(
+                        context.localization.bottomNavBarShoppingCartPriceText,
+                        style: HomePageTextStyles
+                            .bottomNavBarShoppingCartPriceTextStyle,
+                      ),
+                      const SizedBox(
+                        height: 4,
+                      ),
+                      Text(
+                        context.localization.bottomNavBarShoppingCartItemText,
+                        style: HomePageTextStyles
+                            .bottomNavBarShoppingCartItemCountTextStyle,
+                      ),
+                    ],
+                  ),
+                ],
+              ),
+      ),
+    );
+  }
+
+  static Widget getActiveBottomNavBarIcon(String key) {
```

Comment:
you can try switch expressions here
https://dart.dev/language/branches#switch-expressions

```dart
    return switch (key) {
      BottomNavBarKeys.home => AppIcons.homeActiveIcon,
      BottomNavBarKeys.catalogue => AppIcons.catalogueActiveIcon,
      BottomNavBarKeys.favorite => AppIcons.favoriteActiveIcon,
      BottomNavBarKeys.profile => AppIcons.profileActiveIcon,
      _ => const SizedBox(),
    };
 ```